### PR TITLE
bottles: init at 2.1.1

### PR DESCRIPTION
**Pull request by ghost**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/bottles/default.nix [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  '';
+
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
+  postPatch = ''
+    chmod +x build-aux/meson/postinstall.py
+    patchShebangs build-aux/meson/postinstall.py
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/bottles/default.nix:70]:
```suggestion
```

File: pkgs/applications/misc/bottles/default.nix [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+python3Packages.buildPythonApplication rec {
+  pname = "bottles";
+  version = "2.1.1";
+
+  src = fetchFromGitHub {
+    owner = "bottlesdevs";
+    repo = pname;
+    rev = "b0b5a259d3bb5cff89cd6dd7e7be3e6fcf3c7776";
+    sha256 = "1hbjnd06h0h47gcwb1s1b9py5nwmia1m35da6zydbl70vs75imhn";
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/bottles/default.nix:18]:
```suggestion
  };
  
  postPatch = ''
    chmod +x build-aux/meson/postinstall.py
    patchShebangs build-aux/meson/postinstall.py
  '';
```

File: pkgs/applications/misc/bottles/default.nix [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    wrapGAppsHook
+    gettext
+    appstream-glib
+    gobject-introspection
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/bottles/default.nix:27]: Should this be in both inputs?

File: pkgs/applications/misc/bottles/default.nix [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  strictDeps = false; # broken with gobject-introspection setup hook, see https://github.com/NixOS/nixpkgs/issues/56943
+  dontWrapGApps = true; # prevent double wrapping
+
+  preConfigure = ''
+    substituteInPlace build-aux/meson/postinstall.py \
+      --replace "'update-desktop-database'" "'${desktop-file-utils}/bin/update-desktop-database'"
+    substituteInPlace src/runner.py \
+      --replace " {runner}" " ${steam-run-native}/bin/steam-run {runner}"
+    substituteInPlace src/runner.py \
+      --replace " {dxvk_setup}" " ${steam-run-native}/bin/steam-run {dxvk_setup}"
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/bottles/default.nix:60]:
```suggestion
    substituteInPlace src/runner.py \
      --replace " {runner}" " ${steam-run-native}/bin/steam-run {runner}" \
      --replace " {dxvk_setup}" " ${steam-run-native}/bin/steam-run {dxvk_setup}"
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/bottles/default.nix (hunk 3 above)
Comment (bloomvdomino) [reply pkgs/applications/misc/bottles/default.nix:27]: `gobject-introspection`? No, that's no longer needed… :grin:

I've created the derivation initially for an older version of `bottles`, which didn't work without `gobject-introspection` in both.

File: pkgs/applications/misc/bottles/default.nix [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "bottles";
+  version = "2.1.1";
+
+  src = fetchFromGitHub {
+    owner = "bottlesdevs";
+    repo = pname;
+    rev = "b0b5a259d3bb5cff89cd6dd7e7be3e6fcf3c7776";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/bottles/default.nix:16]:
```suggestion
    rev = "v${version}";
```

Comment (bloomvdomino) [reply pkgs/applications/misc/bottles/default.nix]: The releases are tagged without a prefixed 'v'. :smile:

File: pkgs/applications/misc/bottles/default.nix [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "bottles";
+  version = "2.1.1";
+
+  src = fetchFromGitHub {
+    owner = "bottlesdevs";
+    repo = pname;
+    rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/bottles/default.nix:16]:
```suggestion
    rev = version;
```

File: pkgs/applications/misc/bottles/default.nix (hunk 5 above)
Comment (SuperSandro2000) [reply pkgs/applications/misc/bottles/default.nix]: Didn't check that closely.

**Review by Lassulus — APPROVED** [review]

**Pull request merged**